php/example_code/ec2/EC2Service.php

```diff
@@ -0,0 +1,137 @@
+<?php
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+namespace Ec2;
+
+use Aws\Ec2\Ec2Client;
+use Aws\Ec2\Exception\Ec2Exception;
+
+class EC2Service
+{
+    protected Ec2Client $ec2Client;
+    protected string $region;
+
+    public function getRegion(): string
+    {
+        return $this->region;
+    }
+
+    public function __construct($client = null, $region = 'us-west-2', $version = 'latest', $profile = 'default')
+    {
+        if (gettype($client) == Ec2Client::class) {
+            $this->ec2Client = $client;
+            return;
+        }
+        $this->ec2Client = new Ec2Client([
+            'region' => $region,
+            'version' => $version,
+            'profile' => $profile,
+        ]);
+        $this->region = $region;
+        /* Inline declaration example
+        // snippet-start:[php.example_code.ec2.basics.createClient]
+        $ec2Client = new Ec2Client(['region' => 'us-west-2', 'version' => 'latest', 'profile' => 'default']);
+        // snippet-end:[php.example_code.ec2.basics.createClient]
+        */
```

Comment:
What is this for?

Reply:
This shows how to build the client using only inline values. It's used in places in the docs where the writer only needs a quick example of creating the client without the extras added to an example, especially when they're showing how to use a single action that's not covered elsewhere.